diff --git a/app/javascript/application.js b/app/javascript/application.js
index fede9bd..dc58ded 100644
--- a/app/javascript/application.js
+++ b/app/javascript/application.js
@@ -1,6 +1,6 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
 import "@hotwired/turbo-rails"
-import "controllers"
+import "./controllers"
 import * as bootstrap from "bootstrap"
 import "@fortawesome/fontawesome-free"
 
diff --git a/app/javascript/controllers/clipboard_controller.js b/app/javascript/controllers/clipboard_controller.js
index d911a01..25fd3aa 100644
--- a/app/javascript/controllers/clipboard_controller.js
+++ b/app/javascript/controllers/clipboard_controller.js
@@ -10,9 +10,6 @@ export default class extends Controller {
     }
   }
 
-  // copy(){
-  //   navigator.clipboard.writeText(this.sourceTarget.value)
-  // }
   copy(event) {
     event.preventDefault()
     navigator.clipboard.writeText(this.sourceTarget.value)
diff --git a/app/javascript/controllers/practice_controller.js b/app/javascript/controllers/practice_controller.js
index 05c0529..4d171da 100644
--- a/app/javascript/controllers/practice_controller.js
+++ b/app/javascript/controllers/practice_controller.js
@@ -7,11 +7,6 @@ export default class extends Controller {
   connect() {
   }
 
-  // greet(){
-  //   const element = this.nameTarget.value
-  //   console.log(`HEllo, ${element}!`)
-  // }
-
   greet(){
     document.getElementById('demo').innerHTML= (`Hello, ${this.name}!`)
   }
