diff --git a/Gemfile b/Gemfile
index 470b9f1..edf88f4 100644
--- a/Gemfile
+++ b/Gemfile
@@ -77,3 +77,7 @@ gem "cssbundling-rails"
 gem 'country_select', '~> 6.0'
 
 gem "acts_as_list", "~> 1.0"
+
+gem "devise"
+gem "devise-jwt"
+gem 'rack-cors'
\ No newline at end of file
diff --git a/Gemfile.lock b/Gemfile.lock
index 7d2fbeb..b458e7c 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -70,6 +70,7 @@ GEM
       activerecord (>= 4.2)
     addressable (2.8.1)
       public_suffix (>= 2.0.2, < 6.0)
+    bcrypt (3.1.18)
     bindex (0.8.1)
     bootsnap (1.13.0)
       msgpack (~> 1.2)
@@ -97,7 +98,26 @@ GEM
     debug (1.6.3)
       irb (>= 1.3.6)
       reline (>= 0.3.1)
+    devise (4.8.1)
+      bcrypt (~> 3.0)
+      orm_adapter (~> 0.1)
+      railties (>= 4.1.0)
+      responders
+      warden (~> 1.2.3)
+    devise-jwt (0.10.0)
+      devise (~> 4.0)
+      warden-jwt_auth (~> 0.6)
     diff-lcs (1.5.0)
+    dry-auto_inject (0.9.0)
+      dry-container (>= 0.3.4)
+    dry-configurable (0.16.1)
+      dry-core (~> 0.6)
+      zeitwerk (~> 2.6)
+    dry-container (0.11.0)
+      concurrent-ruby (~> 1.0)
+    dry-core (0.9.1)
+      concurrent-ruby (~> 1.0)
+      zeitwerk (~> 2.6)
     erubi (1.11.0)
     factory_bot (6.2.1)
       activesupport (>= 5.0.0)
@@ -123,6 +143,7 @@ GEM
     jbuilder (2.11.5)
       actionview (>= 5.0.0)
       activesupport (>= 5.0.0)
+    jwt (2.5.0)
     loofah (2.19.0)
       crass (~> 1.0.2)
       nokogiri (>= 1.5.9)
@@ -146,11 +167,14 @@ GEM
     nio4r (2.5.8)
     nokogiri (1.13.9-x86_64-linux)
       racc (~> 1.4)
+    orm_adapter (0.5.0)
     public_suffix (5.0.0)
     puma (5.6.5)
       nio4r (~> 2.0)
     racc (1.6.0)
     rack (2.2.4)
+    rack-cors (1.1.1)
+      rack (>= 2.0.0)
     rack-test (2.0.2)
       rack (>= 1.3)
     rails (7.0.4)
@@ -184,6 +208,9 @@ GEM
     regexp_parser (2.6.0)
     reline (0.3.1)
       io-console (~> 0.5)
+    responders (3.0.1)
+      actionpack (>= 5.0)
+      railties (>= 5.0)
     rexml (3.2.5)
     rspec-core (3.12.0)
       rspec-support (~> 3.12.0)
@@ -233,6 +260,13 @@ GEM
     tzinfo (2.0.5)
       concurrent-ruby (~> 1.0)
     unicode_utils (1.4.0)
+    warden (1.2.9)
+      rack (>= 2.0.9)
+    warden-jwt_auth (0.7.0)
+      dry-auto_inject (~> 0.8)
+      dry-configurable (~> 0.13)
+      jwt (~> 2.1)
+      warden (~> 1.2)
     web-console (4.2.0)
       actionview (>= 6.0.0)
       activemodel (>= 6.0.0)
@@ -260,11 +294,14 @@ DEPENDENCIES
   country_select (~> 6.0)
   cssbundling-rails
   debug
+  devise
+  devise-jwt
   factory_bot_rails
   image_processing (~> 1.2)
   importmap-rails
   jbuilder
   puma (~> 5.0)
+  rack-cors
   rails (~> 7.0.2)
   redis (~> 4.0)
   rspec-rails
diff --git a/app/controllers/application_controller.rb b/app/controllers/application_controller.rb
index 09705d1..a01946b 100644
--- a/app/controllers/application_controller.rb
+++ b/app/controllers/application_controller.rb
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+    skip_before_action :verify_authenticity_token
 end
diff --git a/config/application.rb b/config/application.rb
index fe6c31b..21417fb 100644
--- a/config/application.rb
+++ b/config/application.rb
@@ -18,5 +18,9 @@ module Profinder
     #
     # config.time_zone = "Central Time (US & Canada)"
     config.eager_load_paths << Rails.root.join("app/services")
+    config.api_only = true
+    config.session_store :cookie_store, key: '_interslice_session'
+    config.middleware.use ActionDispatch::Cookies
+    config.middleware.use config.session_store, config.session_options
   end
 end
diff --git a/config/environments/development.rb b/config/environments/development.rb
index 8500f45..ded35d7 100644
--- a/config/environments/development.rb
+++ b/config/environments/development.rb
@@ -67,4 +67,5 @@ Rails.application.configure do
 
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
 end
diff --git a/config/routes.rb b/config/routes.rb
index 4bb3aa6..0af8962 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  devise_for :agents, controllers: {
+    sessions: 'agents/sessions',
+    registrations: 'agents/registrations'
+  }
   resources :images
   resources :plants
   resources :users do
diff --git a/db/schema.rb b/db/schema.rb
index 70f5794..9322807 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema[7.0].define(version: 2022_11_27_165855) do
+ActiveRecord::Schema[7.0].define(version: 2022_11_30_175945) do
   create_table "active_storage_attachments", force: :cascade do |t|
     t.string "name", null: false
     t.string "record_type", null: false
@@ -39,6 +39,20 @@ ActiveRecord::Schema[7.0].define(version: 2022_11_27_165855) do
     t.index ["blob_id", "variation_digest"], name: "index_active_storage_variant_records_uniqueness", unique: true
   end
 
+  create_table "agents", force: :cascade do |t|
+    t.string "email", default: "", null: false
+    t.string "encrypted_password", default: "", null: false
+    t.string "reset_password_token"
+    t.datetime "reset_password_sent_at"
+    t.datetime "remember_created_at"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "jti", null: false
+    t.index ["email"], name: "index_agents_on_email", unique: true
+    t.index ["jti"], name: "index_agents_on_jti", unique: true
+    t.index ["reset_password_token"], name: "index_agents_on_reset_password_token", unique: true
+  end
+
   create_table "images", force: :cascade do |t|
     t.string "data_uri"
     t.datetime "created_at", null: false
