diff --git a/app/views/users/_form_modal.html.erb b/app/views/users/_form_modal.html.erb
deleted file mode 100644
index ca66cc4..0000000
--- a/app/views/users/_form_modal.html.erb
+++ /dev/null
@@ -1,13 +0,0 @@
-<%= turbo_frame_tag :remote_modal, target: :_top do %>
-    <div class="modal fade" id="exampleModal" tabindex="-1" aria-labelledby="exampleModalLabel" aria-hidden="true" data-controller="bs-modal">
-        <div class="modal-dialog modal-lg">
-            <div class="modal-content">
-            <div class="modal-header">
-                <h1 class="modal-title fs-5" id="exampleModalLabel">Create New User</h1>
-                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
-            </div>
-              <%= render "form", user: user %>
-            </div>
-        </div>
-    </div>
-<% end %>    
\ No newline at end of file
diff --git a/app/views/users/edit.turbo_stream.erb b/app/views/users/edit.turbo_stream.erb
index 6367e34..73c5e5a 100644
--- a/app/views/users/edit.turbo_stream.erb
+++ b/app/views/users/edit.turbo_stream.erb
@@ -1,3 +1,3 @@
 <%= turbo_stream.replace "remote_modal" do %>
-   <%= render "form_modal", user: @user %>
+   <%= render "modal", user: @user, modal_title: "Edit User", partial_name: "form" %>
 <% end %>
diff --git a/app/views/users/new.turbo_stream.erb b/app/views/users/new.turbo_stream.erb
index 04127a9..37e06b3 100644
--- a/app/views/users/new.turbo_stream.erb
+++ b/app/views/users/new.turbo_stream.erb
@@ -1,3 +1,3 @@
 <%= turbo_stream.replace "remote_modal" do %>
-   <%= render "form_modal", user: @user %>
+   <%= render "modal", user: @user, modal_title: "Create User", partial_name: "form"  %>
 <% end %>
\ No newline at end of file
